pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -3223,5 +3231,22 @@ public void testJoin() {
     Assert.assertEquals(rightSubquery,
         CalciteSqlParser.compileToPinotQuery("SELECT key, COUNT(*) AS b FROM T3 JOIN T4 GROUP BY key"));
     Assert.assertEquals(join.getCondition(), CalciteSqlParser.compileToExpression("T1.key = T2.key"));
+
+    // test for self join queries
+    query = "SELECT T1.a FROM T1 JOIN(SELECT key FROM T1) as self ON T1.key=self.key";
```

Comment:
is `self` a reserved keyword? what's the difference between this test and the one with ` AS T2` ?

Reply:
As per the[ PG documentation ](https://www.postgresql.org/docs/current/sql-keywords-appendix.html), `self` is a non-reserved keyword. There is no difference between `self` and `AS T2`. 
The issue was because of alias(`AS`) usage, not with `self`

Reply:
It is covering; I just added it to make sure `self` has no issues after the function call `extractTableNamesFromNode`.